lib/httpclient/http.rb

```diff
@@ -574,17 +574,19 @@ def reset_pos(io)
       end
 
       def dump_file(io, dev, sz)
+
         buf = ''
         rest = sz
         while rest > 0
           n = io.read([rest, @chunk_size].min, buf)
-          raise ArgumentError.new("Illegal size value: #size returns #{sz} but cannot read") if n.nil?
```

Comment:
https://github.com/bblimke/webmock/blob/da15b1a9bc77f8e2801c136df080d65b7a562685/lib/webmock/http_lib_adapters/httpclient_adapter.rb#L185

In `build_request_signature`, I seem that a file pointer is not reset.